File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/Balloons.java
package com.nopalsoft.ponyrace.game_objects;

import com.esotericsoftware.spine.Skeleton;
import com.nopalsoft.ponyrace.game.TileMapHandler;

public class Balloons extends BaseGameObject {
    public static float TAKEN_ANIMATION_DURATION;
    public float lastStateTime;
    public float stateTime;
    public State state;
    public Skeleton balloonsSkeleton;

    public Balloons(float x, float y, TileMapHandler tileMapHandler) {
        super(x, y);
        stateTime = tileMapHandler.random.nextFloat() * 5f;
        lastStateTime = stateTime;
        state = State.IDLE;
        balloonsSkeleton = new Skeleton(tileMapHandler.game.assetsHandler.globoSkeletonData);
        TAKEN_ANIMATION_DURATION = tileMapHandler.game.assetsHandler.globoTomadaAnim.getDuration();
    }

    public void update(float delta) {
        lastStateTime = stateTime;
        stateTime += delta;
    }

    public void hitPony() {
        state = State.TAKEN;
        stateTime = 0;
    }

    public enum State {
        IDLE,
        TAKEN
    }
}

File: Pony_Racing/core/src/com/nopalsoft/ponyrace/game_objects/Balloons.kt
package com.nopalsoft.ponyrace.game_objects

import com.esotericsoftware.spine.Skeleton
import com.nopalsoft.ponyrace.game.TileMapHandler

class Balloons(x: Float, y: Float, tileMapHandler: TileMapHandler) : BaseGameObject(x, y) {
    @JvmField
    var lastStateTime: Float

    @JvmField
    var stateTime: Float

    @JvmField
    var state: State?

    @JvmField
    var balloonsSkeleton: Skeleton?

    init {
        stateTime = tileMapHandler.random.nextFloat() * 5f
        lastStateTime = stateTime
        state = State.IDLE
        balloonsSkeleton = Skeleton(tileMapHandler.game.assetsHandler.globoSkeletonData)
        TAKEN_ANIMATION_DURATION = tileMapHandler.game.assetsHandler.globoTomadaAnim.duration
    }

    fun update(delta: Float) {
        lastStateTime = stateTime
        stateTime += delta
    }

    fun hitPony() {
        state = State.TAKEN
        stateTime = 0f
    }

    enum class State {
        IDLE,
        TAKEN
    }

    companion object {
        @JvmField
        var TAKEN_ANIMATION_DURATION: Float = 0F
    }
}
